### PR TITLE
Fixing more newly introduced compiler warnings

### PR DESCRIPTION
**Pull request by ashwinstar**
Fix compiler warnings for ext_alloc.c and callers of make_motion_gather_to_*().

### CODE REVIEW

File: src/backend/utils/mmgr/ext_alloc.c [hunk 1]
```diff
@@ -47,7 +48,7 @@ Ext_OptimizerAlloc(size_t size)
 void
 Ext_OptimizerFree(void *ptr)
 {
-	MemoryAccount *account = MemoryAccounting_ConvertIdToAccount(ActiveMemoryAccountId);
+	MemoryAccounting_ConvertIdToAccount(ActiveMemoryAccountId);
```
Comment (hsyuan) [inline src/backend/utils/mmgr/ext_alloc.c]: if it is not used, just remove it.

**Comment by kavinderd** [comment]
Thanks for cleaning up my mess @ashwinstar +1

**Comment by ashwinstar** [comment]
Highly recommend running make and make install with **-s** to (relatively) easily spot the warnings, something I have seen very few folks using.

**Comment by kavinderd** [comment]
Thanks, that's a good tip

**Pull request merged**